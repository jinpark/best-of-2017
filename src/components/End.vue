<template>
  <swiper-slide>
    <div class="end">
      <div class="overlay"></div>
      <video class="bg-video" src="/static/intro.mp4" poster="/static/intro.png" autoplay muted loop></video>
      <div class="video-info">
        <h1 class="title">Honorable Mentions</h1>
        <p>Shows I liked but didn't make it on the list</p>
        <p>âš« - Drama | ðŸ”µ - Comedy | ðŸ”´ - Action | âšª - Cute </p>
        <ul>
          <li>3-Gatsu No Lion âš« </li>
          <li>Aho Girl ðŸ”µ </li>
          <li>Ballroom e Youkoso âš« </li>
          <li>Boku no Hero Academia S2 ðŸ”´ </li>
          <li>Boruto ðŸ”´ </li>
          <li>Demi-chan wa Kataritai âšª </li>
          <li>Gintama ðŸ”µ </li>
          <li>Isekai Shokudou âšª </li>
          <li>Little Witch Academia âšª </li>
          <li>New Game!! âšª </li>
          <li>Owarimonogatari âš« </li>
          <li>Shingeki no Kyojin S2 ðŸ”´ </li>
          <li>Shokugeki no Souma: San no Sara ðŸ”´ </li>
          <li>Shoujo Shuumatsu Ryokou âšª </li>
          <li>Yowamushi Pedal: NEW GENERATION âš« </li>
        </ul>
      </div>
    </div>
  </swiper-slide>
</template>

<script>
export default {
  name: 'End',
  data () {
    return {
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

h1, h2 {
  font-weight: normal;
}
a {
  color: #42b983;
}

video.bg-video {
  position: absolute;
  width: 100vw;
  height: 100vh;
  top: 0;
  left: 0;
  z-index: -2;
  object-fit: cover;
}

.overlay {
    display: block;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    z-index: -1;
    background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAAG0lEQVQYV2NkYGD4z8DAwMgABXAGNgGwSgwVAFbmAgXQdISfAAAAAElFTkSuQmCC);
}

.video-info {
  width: 85vw;
  max-height: 90vh;
  overflow: auto;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background: rgba(0, 0, 0, 0.5);
  padding: 10px;
  color: #ffffff;
}

.title {
  color: #fff;
}
</style>
