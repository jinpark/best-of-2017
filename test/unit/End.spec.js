import End from '@/components/End'
import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

describe('End.vue', () => {
  let wrapper
  const animes = [
    {
      "title": "In This Corner of the World (Film)",
      "japaneseTitle": "この世界の片隅に",
      "filename": "inthiscorneroftheworld",
      "summary": "Set in Hiroshima during World War II, an eighteen-year-old girl gets married and now has to prepare food for her family despite the rationing and lack of supplies. As she struggles with the daily loss of life's amenities she still has to maintain the will to live.",
      "review": "A slice of life story set during WW2 in Japan. Not as soul crushing as 'Grave of the Fireflies' but bring tissues.",
      "youtubeId": "-jBe-uHhlNs"
    },
    {
      "title": "A Silent Voice (Film)",
      "japaneseTitle": "聲の形",
      "filename": "asilentvoice",
      "summary": "A young man is ostracized by his classmates after he bullies a deaf girl to the point where she moves away. Years later, he sets off on a path for redemption.",
      "review": "Everyone has regrets and everyone deals with them differently. What's the right way to redeem yourself?",
      "youtubeId": "nfK6UgLra7g"
    }
  ]
  const horizontalSwiper = {
    on: jest.fn(),
    activeIndex: 1
  }

  beforeEach(() => {
    wrapper = shallow(End, {stubs: {Honorable: '<div class="honorable" />'}})
    wrapper.setComputed({horizontalSwiper})
    wrapper.setMethods({setAnalytics: jest.fn()})
    wrapper.setData({animes})
  })

  it('is a swiper-slide', () => {
    expect(wrapper.is('swiper-slide')).toBe(true)
  })

  it('contains a swiper', () => {
    expect(wrapper.find('swiper').exists()).toBe(true)
  })

  it('is has correct number of honorable slides', () => {
    expect(wrapper.findAll('.honorable')).toHaveLength(animes.length)
  })

  it('matches snapshot', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
