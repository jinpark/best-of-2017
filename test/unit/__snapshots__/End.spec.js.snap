// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End.vue matches snapshot 1`] = `
<swiper-slide>
    <div class="overlay"></div>
    <video src="/static/ending.mp4" autoplay="autoplay" muted="muted" loop="loop" playsinline="" class="bg-video"></video>
    <swiper options="[object Object]">
        <swiper-slide>
            <div class="video-info">
                <h1 class="title">Honorable Mentions</h1>
                <h3 class="japanese-title">Almost on the list but not quite!</h3>
                <h3>Swipe or scroll left to continue</h3>
            </div>
        </swiper-slide>
        <div class="honorable"></div>
        <div class="honorable"></div>
        <div slot="button-prev" class="swiper-button-prev"></div>
        <div slot="button-next" class="swiper-button-next"></div>
    </swiper>
</swiper-slide>
`;
